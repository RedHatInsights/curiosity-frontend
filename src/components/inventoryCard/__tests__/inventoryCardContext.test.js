import {
  context,
  useGetInstancesInventory,
  useOnPageInstances,
  useOnColumnSortInstances
} from '../inventoryCardContext';
import {
  RHSM_API_QUERY_INVENTORY_SORT_DIRECTION_TYPES as SORT_DIRECTION_TYPES,
  RHSM_API_QUERY_INVENTORY_SORT_TYPES as SORT_TYPES
} from '../../../services/rhsm/rhsmConstants';

describe('InventoryCardContext', () => {
  it('should return specific properties', () => {
    expect(context).toMatchSnapshot('specific properties');
  });

  it('should expect specific sort properties', () => {
    expect(SORT_TYPES).toMatchSnapshot('sort properties');
  });

  it('should handle a store response with useGetInstancesInventory', async () => {
    const { result } = await renderHook(
      () =>
        useGetInstancesInventory({
          getInventory: () => () => {},
          useProduct: () => ({ productId: 'lorem' }),
          useDispatch: () => {}
        }),
      {
        state: {
          inventory: {
            instancesInventory: {
              lorem: {
                fulfilled: true,
                data: [{ data: [{ lorem: 'ipsum' }, { dolor: 'sit' }], meta: {} }]
              }
            }
          }
        }
      }
    );

    expect(result).toMatchSnapshot('store response');
  });

  it('should handle variations in instances inventory API responses', async () => {
    const { result: errorResponse } = await renderHook(() =>
      useGetInstancesInventory({
        getInventory: () => () => {},
        useProduct: () => ({ productId: 'lorem' }),
        useDispatch: () => {},
        useSelectorsResponse: () => ({ error: true })
      })
    );

    expect(errorResponse).toMatchSnapshot('inventory, error');

    const { result: pendingResponse } = await renderHook(() =>
      useGetInstancesInventory({
        getInventory: () => () => {},
        useProduct: () => ({ productId: 'lorem' }),
        useDispatch: () => {},
        useSelectorsResponse: () => ({ pending: true })
      })
    );

    expect(pendingResponse).toMatchSnapshot('inventory, pending');

    const { result: cancelledResponse } = await renderHook(() =>
      useGetInstancesInventory({
        getInventory: () => () => {},
        useProduct: () => ({ productId: 'lorem' }),
        useDispatch: () => {},
        useSelectorsResponse: () => ({ cancelled: true })
      })
    );

    expect(cancelledResponse).toMatchSnapshot('inventory, cancelled');

    const { result: fulfilledResponse } = await renderHook(() =>
      useGetInstancesInventory({
        getInventory: () => () => {},
        useProduct: () => ({ productId: 'lorem' }),
        useDispatch: () => {},
        useSelectorsResponse: () => ({ fulfilled: true })
      })
    );

    expect(fulfilledResponse).toMatchSnapshot('inventory, fulfilled');

    const { result: disabledResponse } = await renderHook(() =>
      useGetInstancesInventory({
        isDisabled: true,
        getInventory: () => () => {},
        useProduct: () => ({ productId: 'lorem' }),
        useDispatch: () => {},
        useSelectorsResponse: () => ({ fulfilled: true })
      })
    );

    expect(disabledResponse).toMatchSnapshot('inventory, disabled');
  });

  it('should handle an onPage event', () => {
    const mockDispatch = jest.fn();
    const onPage = useOnPageInstances({
      useDispatch: () => mockDispatch,
      useProduct: () => ({ productId: 'lorem' })
    });

    onPage({ offset: 1, perPage: 5 });
    expect(mockDispatch.mock.calls).toMatchSnapshot('onPage event, dispatch');
    mockDispatch.mockClear();
  });

  it('should handle an onColumnSort event', () => {
    const mockDispatch = jest.fn();
    const onColumnSort = useOnColumnSortInstances({
      sortColumns: { LOREM_IPSUM_COLUMN_ONE: 'loremIpsumColumnOne' },
      useDispatch: () => mockDispatch,
      useProduct: () => ({ productId: 'lorem' })
    });

    onColumnSort(null, { direction: SORT_DIRECTION_TYPES.DESCENDING, id: 'loremIpsumColumnOne' });
    onColumnSort(null, { direction: SORT_DIRECTION_TYPES.ASCENDING, id: 'loremIpsumColumnOne' });

    expect(mockDispatch.mock.calls).toMatchSnapshot('onColumnSort event, dispatch');
    mockDispatch.mockClear();
  });
});
