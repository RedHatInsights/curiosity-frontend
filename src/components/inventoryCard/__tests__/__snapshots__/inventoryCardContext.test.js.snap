// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryCardContext should expect specific sort properties: sort properties 1`] = `
{
  "BILLING_PROVIDER": "billing_provider",
  "CATEGORY": "category",
  "CORES": "Cores",
  "CORE_SECONDS": "Core-seconds",
  "INSTANCE_HOURS": "Instance-hours",
  "LAST_SEEN": "last_seen",
  "NAME": "display_name",
  "NUMBER_OF_GUESTS": "number_of_guests",
  "SOCKETS": "Sockets",
  "STORAGE_GIBIBYTES": "Storage-gibibytes",
  "STORAGE_GIBIBYTE_MONTHS": "Storage-gibibyte-months",
  "TRANSFER_GIBIBYTES": "Transfer-gibibytes",
}
`;

exports[`InventoryCardContext should handle a store response with useGetHostsInventory: store response 1`] = `
{
  "data": [
    {
      "data": [
        {
          "lorem": "ipsum",
        },
        {
          "dolor": "sit",
        },
      ],
      "meta": {},
    },
  ],
  "error": false,
  "fulfilled": true,
  "message": null,
  "pending": false,
  "responses": {
    "id": {},
    "list": [
      {
        "data": [
          {
            "data": [
              {
                "lorem": "ipsum",
              },
              {
                "dolor": "sit",
              },
            ],
            "meta": {},
          },
        ],
        "fulfilled": true,
      },
    ],
  },
}
`;

exports[`InventoryCardContext should handle a store response with useGetInstancesInventory: store response 1`] = `
{
  "data": [
    {
      "data": [
        {
          "lorem": "ipsum",
        },
        {
          "dolor": "sit",
        },
      ],
      "meta": {},
    },
  ],
  "error": false,
  "fulfilled": true,
  "message": null,
  "pending": false,
  "responses": {
    "id": {},
    "list": [
      {
        "data": [
          {
            "data": [
              {
                "lorem": "ipsum",
              },
              {
                "dolor": "sit",
              },
            ],
            "meta": {},
          },
        ],
        "fulfilled": true,
      },
    ],
  },
}
`;

exports[`InventoryCardContext should handle an onColumnSort event for hosts: onColumnSort event, dispatch hosts 1`] = `
[
  [
    [
      {
        "dir": "desc",
        "type": "SET_QUERY_RHSM_HOSTS_INVENTORY_dir",
        "viewId": "lorem",
      },
      {
        "sort": "loremIpsumColumnOne",
        "type": "SET_QUERY_RHSM_HOSTS_INVENTORY_sort",
        "viewId": "lorem",
      },
    ],
  ],
  [
    [
      {
        "dir": "asc",
        "type": "SET_QUERY_RHSM_HOSTS_INVENTORY_dir",
        "viewId": "lorem",
      },
      {
        "sort": "loremIpsumColumnOne",
        "type": "SET_QUERY_RHSM_HOSTS_INVENTORY_sort",
        "viewId": "lorem",
      },
    ],
  ],
]
`;

exports[`InventoryCardContext should handle an onColumnSort event: onColumnSort event, dispatch 1`] = `
[
  [
    [
      {
        "dir": "desc",
        "type": "SET_QUERY_RHSM_HOSTS_INVENTORY_dir",
        "viewId": "lorem",
      },
      {
        "sort": "loremIpsumColumnOne",
        "type": "SET_QUERY_RHSM_HOSTS_INVENTORY_sort",
        "viewId": "lorem",
      },
    ],
  ],
  [
    [
      {
        "dir": "asc",
        "type": "SET_QUERY_RHSM_HOSTS_INVENTORY_dir",
        "viewId": "lorem",
      },
      {
        "sort": "loremIpsumColumnOne",
        "type": "SET_QUERY_RHSM_HOSTS_INVENTORY_sort",
        "viewId": "lorem",
      },
    ],
  ],
]
`;

exports[`InventoryCardContext should handle an onPage event: onPage event, dispatch 1`] = `
[
  [
    [
      {
        "offset": 1,
        "type": "SET_QUERY_RHSM_HOSTS_INVENTORY_offset",
        "viewId": "lorem",
      },
      {
        "limit": 5,
        "type": "SET_QUERY_RHSM_HOSTS_INVENTORY_limit",
        "viewId": "lorem",
      },
    ],
  ],
]
`;

exports[`InventoryCardContext should handle variations in hosts inventory API responses: inventory, cancelled 1`] = `
{
  "data": {},
  "pending": true,
}
`;

exports[`InventoryCardContext should handle variations in hosts inventory API responses: inventory, disabled 1`] = `
{
  "data": {},
  "fulfilled": true,
  "pending": false,
}
`;

exports[`InventoryCardContext should handle variations in hosts inventory API responses: inventory, error 1`] = `
{
  "data": {},
  "error": true,
  "pending": false,
}
`;

exports[`InventoryCardContext should handle variations in hosts inventory API responses: inventory, fulfilled 1`] = `
{
  "data": {},
  "fulfilled": true,
  "pending": false,
}
`;

exports[`InventoryCardContext should handle variations in hosts inventory API responses: inventory, pending 1`] = `
{
  "data": {},
  "pending": true,
}
`;

exports[`InventoryCardContext should handle variations in instances inventory API responses: inventory, cancelled 1`] = `
{
  "data": {},
  "pending": true,
}
`;

exports[`InventoryCardContext should handle variations in instances inventory API responses: inventory, disabled 1`] = `
{
  "data": {},
  "fulfilled": true,
  "pending": false,
}
`;

exports[`InventoryCardContext should handle variations in instances inventory API responses: inventory, error 1`] = `
{
  "data": {},
  "error": true,
  "pending": false,
}
`;

exports[`InventoryCardContext should handle variations in instances inventory API responses: inventory, fulfilled 1`] = `
{
  "data": {},
  "fulfilled": true,
  "pending": false,
}
`;

exports[`InventoryCardContext should handle variations in instances inventory API responses: inventory, pending 1`] = `
{
  "data": {},
  "pending": true,
}
`;

exports[`InventoryCardContext should return specific properties: specific properties 1`] = `
{
  "useGetHostsInventory": [Function],
  "useGetInstancesInventory": [Function],
  "useOnColumnSortHosts": [Function],
  "useOnColumnSortInstances": [Function],
  "useOnPageInstances": [Function],
}
`;
